ddl/db_test.go

```diff
@@ -3416,6 +3420,90 @@ func backgroundExecOnJobUpdatedExported(c *C, s *testDBSuite, hook *ddl.TestDDLC
 	return hook.OnJobUpdatedExported, c3IdxInfo
 }
 
+func (s *testDBSuite) TestColumnModifyingDefinition(c *C) {
+	s.tk = testkit.NewTestKit(c, s.store)
+	s.tk.MustExec("use test")
+	s.tk.MustExec("drop table if exists test2;")
+	s.tk.MustExec("create table test2 (c1 int, c2 int, c3 int default 1, index (c1));")
+	s.tk.MustExec("alter table test2 change c2 a bigint not null;")
+
+	s.tk.MustExec("drop table if exists test2;")
+	s.tk.MustExec("create table test2 (c1 int, c2 int, c3 int default 1, index (c1));")
+	s.tk.MustExec("insert into test2(c2) values (null);")
+	s.testErrorCode(c, "alter table test2 change c2 a bigint not null;", tmysql.WarnDataTruncated)
+}
+
+func (s *testDBSuite) TestModifyColumnRollback(c *C) {
+	s.tk = testkit.NewTestKit(c, s.store)
+	s.mustExec(c, "use test_db")
+	s.mustExec(c, "drop table if exists test2")
+	s.mustExec(c, "create table test2 (c1 int, c2 int, c3 int default 1, index (c1));")
+
+	done := make(chan error, 1)
+	go backgroundExec(s.store, "alter table test2 change c2 c2 bigint not null;", done)
+
+	var checkErr error
+	hook := &ddl.TestDDLCallback{}
+	s.dom.DDL().(ddl.DDLForTest).SetHook(hook)
+	hook.OnJobUpdatedExported = func(job *model.Job) {
+		if checkErr != nil {
```

Comment:
Add a test here, when the PreventNullInsertFlag is set, insert a null value into this table, it must fail.

Reply:
@winkyao Done.